cli/command/init/common.go

```diff
@@ -0,0 +1,147 @@
+package init
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/rs/zerolog/log"
+	"gopkg.in/yaml.v2"
+
+	"github.com/saucelabs/saucectl/internal/config"
+)
+
+
+var configurators = map[string]func(cfg *initConfig) interface{}{
+	"cypress":    configureCypress,
+	"espresso":   configureEspresso,
+	"playwright": configurePlaywright,
+	"puppeteer":  configurePuppeteer,
+	"testcafe":   configureTestcafe,
+	"xcuitest":   configureXCUITest,
+}
+
+var sauceignores = map[string]string {
+	"cypress":    sauceignoreCypress,
+	"playwright": sauceignorePlaywright,
+	"puppeteer":  sauceignorePuppeteer,
+	"testcafe":   sauceignoreTestcafe,
+}
+
+func saveConfigurationFiles(initCfg *initConfig) ([]string, error) {
+	var files []string
+
+	configFormatter, ok := configurators[initCfg.frameworkName]
+	if ok {
+		err := saveSauceConfig(configFormatter(initCfg))
+		if err != nil {
+			return []string{}, err
+		}
+		files = append(files, ".sauce/config.yml")
+	}
+
+	sauceignore, ok := sauceignores[initCfg.frameworkName]
+	if ok {
+		err := saveSauceIgnore(sauceignore)
+		if err != nil {
+			return []string{}, err
+		}
+		files = append(files, ".sauceignore")
+	}
+	return files, nil
+}
+
+func saveSauceConfig(p interface{}) error {
+	fi, err := os.Stat(".sauce")
+	if err != nil && os.IsNotExist(err) {
+		if err = os.Mkdir(".sauce", 0750); err != nil {
+			return err
+		}
+		fi, err = os.Stat(".sauce")
+		if err != nil {
+			return err
+		}
+	}
+	if !fi.IsDir() {
+		return fmt.Errorf(".sauce exists and is not a directory")
+	}
+	fd, err := os.Create(".sauce/config.yml")
+	if err != nil {
+		return err
+	}
+	defer fd.Close()
+	if err = yaml.NewEncoder(fd).Encode(p); err != nil {
+		return err
+	}
+	return nil
+}
+
+func saveSauceIgnore(content string) error {
+	fd, err := os.Create(".sauceignore")
+	if err != nil {
+		return err
+	}
+	defer fd.Close()
+	fd.WriteString(content)
+	return nil
+}
+
+func displaySummary(files []string) {
+	println()
+	log.Info().Msg("The following files have been created:")
```

Comment:
I wouldn't bother using the structured logger for this. Just straight up print 📖  :)

Reply:
That have been changed !